src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/MessagePackHubProtocol.cs

```diff
@@ -0,0 +1,237 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using MsgPack;
+using MsgPack.Serialization;
+
+namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
+{
+    public class MessagePackHubProtocol : IHubProtocol
+    {
+        private const int InvocationMessageType = 1;
+        private const int StreamItemMessageType = 2;
+        private const int CompletionMessageType = 3;
+
+        public bool TryParseMessages(ReadOnlySpan<byte> input, IInvocationBinder binder, out IList<HubMessage> messages)
+        {
+            messages = new List<HubMessage>();
```

Comment:
All it needs is the BinaryMessageParser created here to parse length prefixed messages.